src/jit/gschecks.cpp

```diff
@@ -414,7 +414,18 @@ void Compiler::gsParamsToShadows()
         // Copy some info
 
         var_types type = varTypeIsSmall(varDsc->TypeGet()) ? TYP_INT : varDsc->TypeGet();
-        lvaTable[shadowVar].lvType        = type;
+        lvaTable[shadowVar].lvType = type;
```

Comment:
Not sure that i understood the suggestion.  Are you implying to abstract the logic that copies varDsc fields to shadow param into a separate routine? I don't think that would be valuable, because other gschecks logic no other place has such a need.

VarDsc has many fields but gsParamsToShadow() is selectively copying few of the fields when it creates a shadow lcl var.  Are you implying to copy all fields?  That would need carefully examining whether it makes sense to verbatim copy and is beyond the scope of this fix.